tokio/src/time/interval.rs

```diff
@@ -103,26 +109,179 @@ pub fn interval_at(start: Instant, period: Duration) -> Interval {
+    /// same as it would have if the ticks hadn't been delayed), which results
+    /// in ticks being fired as fast as it can until it is caught up in time to
+    /// where it should be.
+    ///
+    /// This looks something like this:
+    /// ```text
+    /// Expected ticks: |     1     |     2     |     3     |     4     |     5     |     6     |
+    /// Actual ticks:   | work -----|          delay          | work | work | work -| work -----|
+    //  Poll behavior:  |   |       |                         |      |      |       |           |
+    //                  |   |       |                         |      |      |       |           |
+    //           Ready(s)   |       |             Ready(s + 2p)      |      |       |           |
+    //                Pending       |                    Ready(s + 3p)      |       |           |
+    //                   Ready(s + p)                           Ready(s + 4p)       |           |
+    //                                                                  Ready(s + 5p)           |
+    //                                                                              Ready(s + 6p)
+    // * Where `s` is the start time and `p` is the period
+    /// ```
+    ///
+    /// This is the default behavior when [`Interval`] is created with
+    /// [`interval`] and [`interval_at`].
+    ///
+    /// Note: this is the behavior that [`Interval`] exhibited in previous
+    /// versions of Tokio.
+    Burst,
+
+    /// Delay missed ticks to happen at multiples of `period` from when [`tick`]
+    /// was called.
+    ///
+    /// When this strategy is used, [`Interval`] schedules all future ticks to
+    /// happen at a regular `period` from the point when [`tick`] was called.
+    ///
+    /// This looks something like this:
+    /// ```text
+    /// Expected ticks: |     1     |     2     |     3     |     4     |     5     |     6     |
+    /// Actual ticks:   | work -----|          delay          | work -----| work -----| work -----|
+    //  Poll behavior:  |   |       |                         |   |       |           |           |
+    //                  |   |       |                         |   |       |           |           |
+    //           Ready(s)   |       |             Ready(s + 2p)   |       |           |           |
+    //                Pending       |                       Pending       |           |           |
+    //                   Ready(s + p)                     Ready(s + 2p + d)           |           |
+    //                                                                Ready(s + 3p + d)           |
+    //                                                                            Ready(s + 4p + d)
+    // * Where `s` is the start time, `p` is the period, and `d` is the delay
+    /// ```
+    ///
+    /// Note that as a result, the ticks are no longer guaranteed to happen at
+    /// a multiple of `period` from `delay`.
+    ///
+    /// [`tick`]: Interval::tick
+    Delay,
```

Comment:
I don't really understand what this one does.

Reply:
Since you're new to this code, can you help me clarify the documentation so that it's not confusing? The gist of it is this:
```rust
use tokio::time::{interval, Duration};

let mut interval = interval(Duration::from_millis(50));

task_that_takes_more_than_50_millis().await;
// The `Interval` has missed a tick

// Since we have exceeded our timeout, this will resolve immediately
interval.tick().await;

// But this one, rather than also resolving immediately, as might happen
// with the `Burst` behavior (depending on how long the `Interval` was
// delayed), will not resolve until 50ms after the call to `tick` up above.
// That is, in `tick`, when we recognize that we missed a tick, we schedule
// the next tick to happen 50ms (or whatever the `period` is) from right
// then, not from when were were *supposed* to tick
interval.tick().await;
```

Does that make sense?

Reply:
Yes, I understand now.